treeherder/auth/backends.py

```diff
@@ -99,32 +84,31 @@ def authenticate(self, auth_header=None, host=None, port=None):
 
         if result["status"] != "auth-success":
             logger.warning("Error logging in: {}".format(result["message"]))
-            raise TaskclusterAuthenticationFailed(result["message"])
-
-        # TODO: remove this size limit when we upgrade to django 1.10
-        # in Bug 1311967
-        email = self._get_email(result)
-        if len(email) <= 30:
-            username = email
-        else:
-            username = base64.urlsafe_b64encode(
-                hashlib.sha1(smart_bytes(email)).digest()
-                ).rstrip(b'=')[-30:]
+            raise TaskclusterAuthenticationFailedException(result["message"])
+
+        client_id = result["clientId"]
+        email = self._get_email_from_clientid(client_id)
+        username = client_id
 
         try:
```

Comment:
Yeah, this makes sense.  I think it makes the code a bit more read-able as well.  I changed the function names to be a bit better too, I think.

Now I raise immediately if no email is found.